miner/bid_simulator.go

```diff
@@ -593,6 +601,38 @@ func (b *bidSimulator) simBid(interruptCh chan int32, bidRuntime *BidRuntime) {
 		return
 	}
 
+	// fill transactions from mempool
+	if b.config.GreedyMergeTx {
+		delay := b.engine.Delay(b.chain, bidRuntime.env.header, &b.delayLeftOver)
+		if delay != nil && *delay > 0 {
+			log.Debug("BidSimulator: GreedyMergeTx stopTimer", "block", bidRuntime.env.header.Number,
+				"header time", time.Until(time.Unix(int64(bidRuntime.env.header.Time), 0)),
+				"commit delay", *delay, "DelayLeftOver", b.delayLeftOver)
+
+			stopTimer := time.NewTimer(*delay)
+
+			bidTxsSet := mapset.NewSet[common.Hash]()
+			for _, tx := range bidRuntime.bid.Txs {
+				bidTxsSet.Add(tx.Hash())
+			}
+
+			fillErr := b.bidWorker.fillTransactions(interruptCh, bidRuntime.env, stopTimer, bidTxsSet, params.TxGas+params.PayBidTxGasLimit)
```

Comment:
both `params.TxGas` and `params.PayBidTxGasLimit` are constant, so I think no need pass it as a parameter, it is a long way to go.  

And the patch does not fix the problem:
```
		if env.gasPool.Gas() < minGas {
			log.Trace("Not enough gas for further transactions", "have", env.gasPool, "want", minGas)
			signal = commitInterruptOutOfGas
			break
		}
```
It is still possible that `env.gasPool.Gas()` is smaller than 25000